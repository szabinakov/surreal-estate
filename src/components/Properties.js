import React, { useState, useEffect } from 'react'
import PropertyCard from './PropertyCard'
import axios from 'axios'
import '../styles/Properties.css'
import Alert from '../components/Alert'

const Properties = () => {

    const initialState = {
        alert: {
            message: ''
        }
    }
    const [properties, setProperties] = useState([])
    const [alert, setAlert] = useState(initialState.alert)

    useEffect(()=>{
        axios.get('http://localhost:4000/api/v1/PropertyListing')
        .then(response => {
            setProperties(response.data)
        })
        .catch(() =>{
            setAlert({message: 'Something went wrong!'})
        })
    },[])

    return (
        <>
        <h2>Properties</h2>
        {alert.message && <Alert message={alert.message}/>} 
        <PropertyCard details={properties}/>
        </>
    )
}

export default Properties