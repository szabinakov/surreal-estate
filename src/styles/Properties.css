.Properties {
    border-top-style: groove;
    border-color: rgba(0, 0, 0, 0.096);
    padding-top: 10px;
    margin-top: 80px;
    display: flex;
    flex-direction: column;
}

.PropertyCards {
    border: solid salmon;
}
