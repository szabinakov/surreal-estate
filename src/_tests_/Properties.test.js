import React from 'react';
import { render } from '@testing-library/react';
import Properties from '../components/Properties';

describe('Properties', () => {
    const {asFragment} = render (<Properties/>)

    it('renders correctly', () => {

        expect(asFragment()).toMatchSnapshot()
    })
})