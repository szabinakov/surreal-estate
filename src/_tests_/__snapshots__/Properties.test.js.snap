// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<DocumentFragment>
  <h2>
    Properties
  </h2>
  <div
    class="PropertyContainer"
  />
</DocumentFragment>
`;
